### PR TITLE
test(engine): fix start etcd test failure because of duplicated free port

### LINKED ISSUE #5515: Unstable unit test TestStartEtcdTimeout in dataflow engine
### Which jobs are flaking?

https://github.com/pingcap/tiflow/runs/6548240417?check_suite_focus=true

### Which test(s) are flaking?

TestStartEtcdTimeout 

### Jenkins logs or GitHub Actions link

```go
--- FAIL: TestStartEtcdTimeout (0.01s)
    etcd_test.go:66: 
        	Error Trace:	etcd_test.go:66
        	Error:      	Error message not equal:
        	            	expected: "[DFLOW:ErrMasterStartEmbedEtcdFail]start embed etcd timeout 100ms"
        	            	actual  : "[DFLOW:ErrMasterStartEmbedEtcdFail]failed to start embed etcd: initial cluster test-start-etcd-timeout-1=[http://127.0.0.1:43055,test-start-etcd-timeout-2=http://127.0.0.1:43055](http://127.0.0.1:43055%2Ctest-start-etcd-timeout-2%3Dhttp//127.0.0.1:43055) has duplicate url"
        	Test:       	TestStartEtcdTimeout
```

### Anything else we need to know

- Does this test exist for other branches as well?

- Has there been a high frequency of failure lately?

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: close #5515

### What is changed and how it works?

Network raw listen on `:0` could get a duplicated free port, just retry several times to get a different port.

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by liuzix — APPROVED** [review]

File: engine/servermaster/etcd_test.go [hunk 1]
```diff
@@ -53,7 +53,11 @@ func TestStartEtcdTimeout(t *testing.T) {
 	cfgCluster.Name = name1
 	cfgCluster.DataDir = dir
 	peer1 := allocTempURL(t)
-	peer2 := allocTempURL(t)
+	peer2 := peer1
```
Comment (maxshuang) [inline engine/servermaster/etcd_test.go:56]: I think the main reason we get this error is `peer1 is not used when allocate peer2`.

Comment (maxshuang) [reply engine/servermaster/etcd_test.go:56]: Need change the pr discription  before merge this pr.

**Review by maxshuang — APPROVED** [review]

**Pull request merged**